actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -77,6 +77,10 @@ module RequestForgeryProtection
       config_accessor :log_warning_on_csrf_failure
       self.log_warning_on_csrf_failure = true
 
+      # Controls whether the Origin header is checked in addition to the CSRF token.
+      config_accessor :forgery_protection_origin_check
+      self.forgery_protection_origin_check = false
+
```

Comment:
Seems this *might* be safely enabled by default without breaking back compat since we're permitting requests from older browsers without an Origin header. When an Origin is provided, it must match. Could see this breaking folks who deploy with a reverse proxy that rewrites the Host header, so every request would be a false-positive CSRF.